python/dgl/nn/pytorch/conv/cugraph_gatconv.py

```diff
@@ -5,24 +5,26 @@
 import torch
 from torch import nn
 
+from .cugraph_base import CuGraphBaseConv
+
 try:
-    from pylibcugraphops import make_fg_csr, make_mfg_csr
-    from pylibcugraphops.torch.autograd import mha_gat_n2n as GATConvAgg
+    from pylibcugraphops.pytorch import SampledCSC, StaticCSC
+    from pylibcugraphops.pytorch.operators import mha_gat_n2n as GATConvAgg
+
+    HAS_PYLIBCUGRAPHOPS = True
 except ImportError:
-    has_pylibcugraphops = False
-else:
-    has_pylibcugraphops = True
+    HAS_PYLIBCUGRAPHOPS = False
 
 
-class CuGraphGATConv(nn.Module):
+class CuGraphGATConv(CuGraphBaseConv):
     r"""Graph attention layer from `Graph Attention Networks
     <https://arxiv.org/pdf/1710.10903.pdf>`__, with the sparse aggregation
     accelerated by cugraph-ops.
 
     See :class:`dgl.nn.pytorch.conv.GATConv` for mathematical model.
 
     This module depends on :code:`pylibcugraphops` package, which can be
-    installed via :code:`conda install -c nvidia pylibcugraphops>=23.02`.
+    installed via :code:`conda install -c nvidia pylibcugraphops>=23.04`.
```

Comment:
Can you try to install it again the quotes around the relationship?
`conda install -c nvidia 'pylibcugraphops>=23.04'`

> Have you uploaded the distribution to Conda?

Yes, it is here: https://anaconda.org/nvidia/pylibcugraphops

> Do I need to separately install CuGraph?

No, pylibcugraphops does not depend on CuGraph.

> Does it have requirements for CUDA version?

Any CUDA toolkit >=11 should work.

Reply:
With the quotes added, I now encountered

```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.31=0
  - feature:|@/linux-64::__glibc==2.31=0
  - pylibcugraphops[version='>=23.04'] -> __glibc[version='>=2.17,<3.0.a0']

Your installed version is: 2.31

Note that strict channel priority may have removed packages required for satisfiability.
```

Do you have any clues?

Reply:
The __glibc version looks fine, so I guess the culprit might be the python version. pylibcugraphops requires python>=3.10. Can you check that?

Here is a simple conda environment yaml file:
```yaml
name: pylibcugraphops_test
channels:
  - pytorch
  - conda-forge
  - nvidia
  - rapidsai
dependencies:
  - python>=3.10
  - pylibcugraphops=23.04
  - jupyter
  - cmake
  - pip
  - pytest
  - pytorch>=2.0
  - pytorch-cuda=11.8
```

Reply:
This can be potentially the reason as we support various Python versions before 3.10, as documented [here](https://www.dgl.ai/pages/start.html). Could you update the doc? Also, is there a way to raise a warning message based on Python version?

Reply:
I encountered another issue in building from source #5629 . I'll try it after addressing this issue.

Reply:
Thanks for the feedback. I just updated the docs to specify Python version requirements for pylibcugraphops=23.04. 

> Also, is there a way to raise a warning message based on Python version?

IMO, we should delegate the task of resolving dependencies' versions to package managers like conda.

Reply:
Yes, but this is tricky as a Conda environment is for a specific Python version.

Reply:
@tingyu66 I tried with a Conda environment with Python 3.10.10, PyTorch 2.0 and CUDA 11.7, but I still encountered similar issues below.

```
(dgl-dev-gpu) ubuntu@ip-172-31-17-164:~/cugraph_230504/dgl/python$ conda install -c nvidia pylibcugraphops=23.04
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: -
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.27=0
  - feature:|@/linux-64::__glibc==2.27=0
  - pylibcugraphops=23.04 -> __glibc[version='>=2.17,<3.0.a0']

Your installed version is: 2.27
```

Reply:
Same error reported here: https://github.com/dmlc/dgl/issues/5672.

Can you try `conda install -c nvidia -c conda-forge -c rapidsai pylibcugraphops=23.04` instead?

Reply:
@tingyu66 This works. However, it installed `pylibcugraphops-23.0` and I encountered 

```
ModuleNotFoundError: CuGraphSAGEConv requires pylibcugraphops=23.04. Install via `conda install -c nvidia 'pylibcugraphops=23.04'`.pylibcugraphops requires Python 3.8 or 3.10.
```

when trying the example.

Reply:
That's a bit weird, since one should only see that error when `HAS_PYLIBCUGRAPHOPS` is False:
https://github.com/dmlc/dgl/blob/41baa0e4483b493e74c4dee6dc67abf6d120a1cc/python/dgl/nn/pytorch/conv/cugraph_sageconv.py#LL9C1-L15C32

I tested it on my end using the "dgl-dev-gpu" conda environment and installed pylibcugraphops on top. The example worked, see: https://github.com/dmlc/dgl/issues/5672#issuecomment-1543234453.

What is your output for `conda list pylibcugraphops`?

Reply:
@tingyu66 My bad. It works now. I might did something wrong. Thanks!